File: blog/src/main/java/cn/pompip/myblog/control/UserController.java
package cn.pompip.myblog.control;

import cn.pompip.myblog.server.UserService;
import cn.pompip.myblog.utils.NeedToken;
import org.springframework.http.HttpHeaders;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import javax.servlet.http.HttpServletRequest;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

@RestController
@RequestMapping("/user")
//@CrossOrigin
public class UserController {

    @Resource
    UserService userService;

    @PostMapping("/logon")
    public String register() {
        return "register";
    }

    @PostMapping("/login")
    public Map<String, String> login(@RequestParam String name, @RequestParam String password) throws Exception {
        Logger.getLogger("").info("name:" + name + " pwd:" + password);
        if ("chong".equals(name) && password.equals("314159")) {
            String token = userService.getToken(name);
            Map<String, String> tokenResponse = new HashMap<>();
            userService.addUser(token, "chong");
            tokenResponse.put("token", token);
            return tokenResponse;
        } else {
            throw new RuntimeException();
        }
    }

    @NeedToken
    @PostMapping("/refresh")
    public Map<String, String> refreshToken(@RequestHeader String token) throws Exception {
        Logger.getLogger("").info("token:" + token);
        String newToken = userService.refreshToken(token);
        Map<String, String> tokenResponse = new HashMap<>();
        userService.addUser(token, "chong");
        tokenResponse.put("token", newToken);
        return tokenResponse;

    }

    @PostMapping("/logout")
    public int logout(HttpServletRequest request) {
        String headToken = request.getHeader("token");
        userService.remoteUser(headToken);
        return 1;
    }

}

File: src/main/kotlin/cn/pompip/myblog/control/UserController.kt
package cn.pompip.myblog.control

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import javax.servlet.http.HttpSession

@Controller
@RequestMapping("/user")
class UserController {

    @GetMapping("/login")
    fun login() = "login"

    @ResponseBody
    @PostMapping("/login")
    fun login(name: String, pwd: String, session: HttpSession) :Int{
        if ("chong" == name && pwd == "314159") {
            session.setAttribute("user", "chong")
            return 1
        }else{
            return 0
        }
    }
    @ResponseBody
    @PostMapping("/logout")
    fun logout( session: HttpSession) :Int{
            session.removeAttribute("user")
            return 1
    }

}